From 24d46521ff7f113d3a3f6cf94d492aad9d67034b Mon Sep 17 00:00:00 2001
From: Gunaal R <quic_gunar@quicinc.com>
Date: Mon, 30 Dec 2024 14:51:36 +0530
Subject: [PATCH] QSDK:CP:wifi-scripts: Fix wireless config generation for
 different phy

Currently in mac80211.uc if single wiphy have more than one phy,
radio_exists function will return and config for remaining phy will
not be generated.

Fix it by skipping radio_exists function and handled it
to generate config for remaining phy.

Change-Id: Ie18bbfe2fb7725b68a9d03895b640b0f4c8589dc
Signed-off-by: Gunaal R <quic_gunar@quicinc.com>
---
 .../config/wifi-scripts/files/lib/wifi/mac80211.uc    | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/package/network/config/wifi-scripts/files/lib/wifi/mac80211.uc b/package/network/config/wifi-scripts/files/lib/wifi/mac80211.uc
index fe564a0d67e0..4496ac9a4056 100644
--- a/package/network/config/wifi-scripts/files/lib/wifi/mac80211.uc
+++ b/package/network/config/wifi-scripts/files/lib/wifi/mac80211.uc
@@ -184,8 +184,10 @@ for (let phy_name, phy in board.wlan) {
 		return;
 
 	let macaddr = trim(readfile(`/sys/class/ieee80211/${phy_name}/macaddress`));
-	if (radio_exists(phy.path, macaddr, phy_name))
-		return;
+	if (radio_exists(phy.path, macaddr, phy_name)) {
+		idx++;
+		continue;
+	}
 
 	id = `phy='${phy_name}'`;
 	if (match(phy_name, /^phy[0-9]/))
@@ -203,11 +205,10 @@ for (let phy_name, phy in board.wlan) {
 			generate_config(info, name, single_wiphy, id, radio_idx);
 		}
 	} else {
-		while (config[`radio${idx}`])
-			idx++;
-		name = "radio" + idx++;
+		name = "radio" + idx;
 		generate_config(info, name, single_wiphy, id, NULL);
 	}
+	idx++;
 	commit = true;
 }
 
-- 
2.34.1

